src/Trace/TraceClient.php

```diff
@@ -0,0 +1,199 @@
+     *
+     *     @type string $projectId The project ID from the Google Developer's
+     *           Console.
+     *     @type CacheItemPoolInterface $authCache A cache used storing access
+     *           tokens. **Defaults to** a simple in memory implementation.
+     *     @type array $authCacheOptions Cache configuration options.
+     *     @type callable $authHttpHandler A handler used to deliver Psr7
+     *           requests specifically for authentication.
+     *     @type callable $httpHandler A handler used to deliver Psr7 requests.
+     *           Only valid for requests sent over REST.
+     *     @type array $keyFile The contents of the service account credentials
+     *           .json file retrieved from the Google Developer's Console.
+     *           Ex: `json_decode(file_get_contents($path), true)`.
+     *     @type string $keyFilePath The full path to your service account
+     *           credentials .json file retrieved from the Google Developers
+     *           Console.
+     *     @type int $retries Number of retries for a failed request.
+     *           **Defaults to** `3`.
+     *     @type array $scopes Scopes to be used for the request.
+     * }
+     */
+    public function __construct(array $config = [])
+    {
+        if (!isset($config['scopes'])) {
+            $config['scopes'] = [self::FULL_CONTROL_SCOPE];
+        }
+
+        $this->connection = new Rest($this->configureAuthentication($config));
+    }
+
+    /**
+     * Returns the projectId for this client.
+     *
+     * @return string
+     */
+    public function projectId()
+    {
+        return $this->projectId;
+    }
+
+    /**
+     * Sends a Trace log in a simple fashion.
+     *
+     * @param  Trace $trace The trace log to send.
+     * @return bool
+     * @throws ServiceException
+     */
+    public function insertTrace(Trace $trace)
+    {
+        return $this->insertTraceBatch([$trace]);
+    }
+
+    /**
+     * Sends multiple Trace logs in a simple fashion.
+     *
+     * @param Trace[] $traces The trace logs to send.
+     * @return bool
+     * @throws ServiceException
+     */
+    public function insertTraceBatch(array $traces)
+    {
+        // throws ServiceException on failure
+        $this->connection->patchTraces([
+            'projectId' => $this->projectId,
+            'traces' => array_map(function ($trace) {
+                return $trace->info();
+            }, $traces)
+        ]);
+        return true;
+    }
+
+    /**
+     * Fetch a single trace by traceId.
+     *
+     * @param  string $traceId The ID of the trace to return
+     * @return Trace
+     * @throws ServiceException
+     */
+    public function getTrace($traceId)
+    {
+        $trace = $this->connection->getTrace([
+            'projectId' => $this->projectId,
+            'traceId' => $traceId
+        ]);
+
+        if (empty($trace)) {
+            throw new NotFoundException('Trace ID does not exist', 404);
```

Comment:
Ok, filed a bug on this for the Trace API team. If it can be fixed, that API call will return a 404 and should throw the `NotFoundException` from the connection. At that point, we could remove this empty check.